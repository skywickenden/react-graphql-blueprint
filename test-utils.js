import React from 'react';
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from 'react-router-dom';
import { QueryMock } from 'graphql-query-test-mock';

// Required when mocking due to babel
import regeneratorRuntime from "regenerator-runtime";

// Required by nock for mocking graphql requests.
global.fetch = require('node-fetch');

const queryMock = new QueryMock();
beforeEach(() => {
  queryMock.setup("http://localhost:3000/graphql"); // Variable containing the URL for your GraphQL API. This must match what you"re making requests to in your client code.
});

const wrapper = ({children}) => (
  <Router>{children}</Router>
);

const customRender = (ui, options) =>
  render(ui, { wrapper, ...options });
  
export * from '@testing-library/react';

export { customRender as render, queryMock };
