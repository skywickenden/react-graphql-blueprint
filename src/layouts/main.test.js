import React from "react";
import { render } from "../../test-utils";
import Main from "./main";

import { collect } from "linaria/server";
import fs from "fs";
import path from "path";
const css = fs.readFileSync(path.resolve("./.linaria-cache/src/layouts/main.linaria.css"), "utf8");

it("Renders and has text content", () => {
  const { getByText, asFragment, getByTestId  } = render(<Main />);

  // Only need to test elements with styles attached
  // All HTML is tested by the snapshot
  const title = getByText("React GraphQL Blueprint");

  const baseElement = getByTestId("base");
  const { critical: baseCSS } = collect(baseElement, css);
  expect(baseCSS).toEqual(expect.stringContaining("font-family:Arial,Helvetica,sans-serif;"));
  
  const { critical: titleCSS } = collect(title, css);
  expect(titleCSS).toEqual(expect.stringContaining("color:#872258;"));
  expect(titleCSS).toEqual(expect.stringContaining("font-size:30px;"));

  const homeLink = getByText("Home");
  const { critical: homeLinkCSS } = collect(homeLink, css);
  expect(homeLinkCSS).toEqual(expect.stringContaining("margin:2px 5px;"));

  const fooLink = getByText("Foo");
  const { critical: fooLinkCSS } = collect(fooLink, css);
  expect(fooLinkCSS).toEqual(expect.stringContaining("margin:2px 5px;"));

  const { critical: navCSS } = collect(homeLink.parent, css);
  expect(navCSS).toEqual(expect.stringContaining("background-color:#eeeeff;"));
  expect(navCSS).toEqual(expect.stringContaining("padding:5px;"));

  // This is here as an alternative example for how to test
  // However snapshots do not test the css and tend to be quite brittle.
  expect(asFragment()).toMatchSnapshot();
});