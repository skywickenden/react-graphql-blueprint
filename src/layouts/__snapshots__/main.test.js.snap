// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders and has text content 1`] = `
<DocumentFragment>
  <div>
    <header>
      <h1
        class="t1gffk2o"
      >
        React GraphQL Blueprint
      </h1>
      <nav
        class="nkso86p"
      >
        <a
          class="nn3ty9u"
          href="/"
        >
          Home
        </a>
        <a
          class="nn3ty9u"
          href="/foo"
        >
          Foo
        </a>
      </nav>
    </header>
    <main />
  </div>
</DocumentFragment>
`;
